substrate/frame/revive/src/weights.rs

```diff
@@ -975,85 +975,85 @@ impl<T: frame_system::Config> WeightInfo for SubstrateWeight<T> {
+			// Standard Error: 3
+			.saturating_add(Weight::from_parts(1_454, 0).saturating_mul(n.into()))
 	}
 	/// The range of component `n` is `[0, 262144]`.
 	fn seal_hash_blake2_128(n: u32, ) -> Weight {
 		// Proof Size summary in bytes:
 		//  Measured:  `0`
 		//  Estimated: `0`
-		// Minimum execution time: 582_000 picoseconds.
-		Weight::from_parts(2_564_984, 0)
+		// Minimum execution time: 617_000 picoseconds.
+		Weight::from_parts(3_212_292, 0)
 			// Standard Error: 2
-			.saturating_add(Weight::from_parts(1_456, 0).saturating_mul(n.into()))
+			.saturating_add(Weight::from_parts(1_452, 0).saturating_mul(n.into()))
 	}
 	/// The range of component `n` is `[0, 261889]`.
 	fn seal_sr25519_verify(n: u32, ) -> Weight {
 		// Proof Size summary in bytes:
 		//  Measured:  `0`
 		//  Estimated: `0`
-		// Minimum execution time: 47_424_000 picoseconds.
-		Weight::from_parts(31_409_241, 0)
-			// Standard Error: 10
-			.saturating_add(Weight::from_parts(5_162, 0).saturating_mul(n.into()))
+		// Minimum execution time: 43_289_000 picoseconds.
+		Weight::from_parts(33_267_379, 0)
+			// Standard Error: 11
+			.saturating_add(Weight::from_parts(5_021, 0).saturating_mul(n.into()))
 	}
 	fn seal_ecdsa_recover() -> Weight {
 		// Proof Size summary in bytes:
 		//  Measured:  `0`
 		//  Estimated: `0`
-		// Minimum execution time: 45_744_000 picoseconds.
-		Weight::from_parts(47_340_000, 0)
+		// Minimum execution time: 46_146_000 picoseconds.
+		Weight::from_parts(47_929_000, 0)
 	}
 	fn seal_ecdsa_to_eth_address() -> Weight {
 		// Proof Size summary in bytes:
 		//  Measured:  `0`
 		//  Estimated: `0`
-		// Minimum execution time: 12_258_000 picoseconds.
-		Weight::from_parts(12_400_000, 0)
+		// Minimum execution time: 12_359_000 picoseconds.
+		Weight::from_parts(12_538_000, 0)
 	}
 	/// Storage: `Revive::CodeInfoOf` (r:1 w:1)
 	/// Proof: `Revive::CodeInfoOf` (`max_values`: None, `max_size`: Some(96), added: 2571, mode: `Measured`)
 	fn seal_set_code_hash() -> Weight {
 		// Proof Size summary in bytes:
 		//  Measured:  `196`
 		//  Estimated: `3661`
-		// Minimum execution time: 9_637_000 picoseconds.
-		Weight::from_parts(9_922_000, 3661)
+		// Minimum execution time: 9_844_000 picoseconds.
+		Weight::from_parts(10_243_000, 3661)
 			.saturating_add(T::DbWeight::get().reads(1_u64))
 			.saturating_add(T::DbWeight::get().writes(1_u64))
 	}
-	/// The range of component `r` is `[0, 5000]`.
+	/// The range of component `r` is `[0, 1000]`.
 	fn instr(r: u32, ) -> Weight {
 		// Proof Size summary in bytes:
 		//  Measured:  `0`
 		//  Estimated: `0`
-		// Minimum execution time: 7_505_000 picoseconds.
-		Weight::from_parts(9_608_207, 0)
-			// Standard Error: 152
-			.saturating_add(Weight::from_parts(73_332, 0).saturating_mul(r.into()))
+		// Minimum execution time: 11_617_000 picoseconds.
+		Weight::from_parts(15_157_062, 0)
+			// Standard Error: 609
+			.saturating_add(Weight::from_parts(173_172, 0).saturating_mul(r.into()))
```

Comment:
`173ns` per loop iteration. If we'd take this as instruction weight we are probably cooked. Maybe just devide by the 4 instructions that are part of the loop? Even though the cache miss will be much more expensive.